python/plugins/processing/algs/gdal/gdalcalc.py

```diff
@@ -105,24 +172,32 @@ def group(self):
     def groupId(self):
         return 'rastermiscellaneous'
 
+    def commandName(self):
+        if isWindows():
+            return 'gdal_calc'
+        else:
+            return 'gdal_calc.py'
+
     def getConsoleCommands(self, parameters, context, feedback, executing=True):
-        out = self.getOutputValue(self.OUTPUT)
-        extra = self.getParameterValue(self.EXTRA)
-        if extra is not None:
-            extra = str(extra)
-        #debug = self.getParameterValue(self.DEBUG)
-        formula = self.getParameterValue(self.FORMULA)
-        noData = self.getParameterValue(self.NO_DATA)
-        if noData is not None:
-            noData = str(noData)
+        if not parameters:
```

Comment:
Shouldn't be needed

Reply:
otherwise would fail test https://github.com/qgis/QGIS/blob/master/python/plugins/processing/tests/GdalAlgorithmsTest.py#L112
or the test should be adapted to consider other type of processing exceptions

Reply:
If you implement the other changes I've listed here then that test should pass ok

Reply:
ok I see, tnx